File: src/examples/go/hello.go
export default {
  id: 'hello-go',
  language: 'go',
  title: 'Hello Go (Goroutines)',
  description: 'Go hello world demonstrating goroutines and channels',
  difficulty: 'intermediate',
  category: 'async',
  tags: ['concurrency', 'goroutines', 'channels'],
  code: `package main

import (
  "fmt"
  "time"
)

func greet(msg string, done chan bool) {
  for i := 0; i < 3; i++ {
    fmt.Println(msg, i)
    time.Sleep(100 * time.Millisecond)
  }
  done <- true
}

func main() {
  done := make(chan bool)
  go greet("Hello from goroutine", done)
  go greet("Concurrent execution", done)

  <-done
  <-done
  fmt.Println("All goroutines finished")
}`
};

File: src/examples/rust/hello.rs
export default {
  id: 'hello-rust',
  language: 'rust',
  title: 'Hello Rust (Ownership Demo)',
  description: 'Rust hello world showcasing ownership and borrowing',
  difficulty: 'intermediate',
  category: 'basics',
  tags: ['ownership', 'borrowing', 'basics'],
  code: `fn main() {
  let s1 = String::from("hello");
  let len = calculate_length(&s1); // borrow
  println!("'{}' has length {}", s1, len);

  let s2 = take_and_return(s1);    // move
  // println!("{}", s1);           // would not compile
  println!("s2 = {}", s2);
}

fn calculate_length(s: &String) -> usize {
  s.len()
}

fn take_and_return(s: String) -> String {
  println!("Took ownership of: {}", s);
  s
}`
};
